app/javascript/react/ansibleCatalog/ansibleCatalogItemForm.jsx

```diff
@@ -0,0 +1,375 @@
+        config_info: {
+            provision,
+            retirement
+        }
+      }
+    };
+    const url = '/api/service_templates/' + (!isNew? catalogItemFormId:'');
+    const msg = isNew ? 'added': 'updated';
+    API.post(url, catalogItem).then(result => {
+      window.add_flash(__(`Catalog item was ${msg} successfully.`),'success');
+    }).catch(() => {
+      window.add_flash(__(`Catalog item failed to be ${msg}.`),'error');
+    });
+  }
+  render() {
+    const {region} = this.props;
+    const {ansibleCatalog, duplicateDropdowns} = this.props;
+    const dropdowns = ansibleCatalog.dropdowns || {};
+    const formMutators = {
+      ...arrayMutators,
+    copyValues: (args, state, utils) => {
+      ANSIBLE_FIELDS.forEach((field) => {
+       utils.changeValue(state, `retirement_${field}`, () => state.formState.values[`provisioning_${field}`])
+      });
+      duplicateDropdowns(['playbooks', 'cloudCredentials']);
+    }};
+    const formFields = [
+      {
+        component: FinalFormField,
+        name: 'name',
+        label: __('Name'),
+        validateOnMount: true,
+        validate: required({ msg: __('Required') })
+      },
+      {
+        component: FinalFormField,
+        name: 'description',
+        id: 'description',
+        label: __('Description'),
+      },
+      {
+        name: 'display',
+        component: FinalFormSwitch,
+        label: __('Display in Catalog'),
+        onText: 'Yes',
+        offText: 'No',
+      },
+      {
+        condition: {when: 'display', is: true},
+        name: 'long_description',
+        label: __('Long Description'),
+        component: FinalFormTextArea
+      },
+      {
+        options: dropdowns.catalogs || [],
+        label: __('Catalog'),
+        component: FinalFormSelect,
+        name: 'service_template_catalog_id',
+        searchable: true
+      }
+    ];
+  const dialogOptions = [
+    {label: __('Use Existing'), value: 'existing'},
+    {label: __('Create New'), value: 'new'}
+  ];
+  const bsClass = 'nav nav-tabs nav-tabs-pf';
+
+  const getResourceOptions = (values) => {
+    if (values['retirement_playbook_id']) {
+      return PLAYBOOK_OPTIONS;
+    }
+
+    return [
+      {label:__('No') ,value:'no_without_playbook'},
+      {label:__('Yes') ,value:'yes_without_playbook'}
+    ]
+  }
+  const { isLoading, initialValues, isNew } = this.state;
+  return (isLoading? ( <Spinner loading size="lg" />) :
+    (<Form
+      initialValues={initialValues}
+      onSubmit={this.submitForm}
+      mutators={{
+        ...formMutators
+      }}
+      render={({ invalid, values,submitting,pristine, handleSubmit, form: { change,reset, mutators: { push, copyValues } } }) => (
+        <PfForm horizontal onSubmit={handleSubmit}>
+        <div id="flash_msg_div"/>
+          {
+            formFields.map(fields => {
+              if (fields['condition'])
+              {
+                if (values[fields.condition['when']] !== fields.condition.is) {
```

Comment:
I think you can use the  `Condition` component here.

Reply:
Tried using the condition component and it was generating an error that was very hard to try and debug.  This method resulted in no errors and the condition field rendering as expected.